plugin/trino-hive/src/main/java/io/trino/plugin/hive/HiveConfig.java

```diff
@@ -520,6 +521,19 @@ public HiveConfig setCreateEmptyBucketFiles(boolean createEmptyBucketFiles)
         return this;
     }
 
+    public boolean isDeleteSchemaLocationsFallback()
+    {
+        return this.deleteSchemaLocationsFallback;
+    }
+
+    @Config("hive.delete-schema-locations-fallback")
```

Comment:
i am not sure of the name, but couldn't find a better one.
@losipiuk @alexjo2144 ideas?

Reply:
I originally had "delete schema locations by default," but it really is meant only as a fallback.

Reply:
Maybe `delete-schema-locations-as-fallback` would be very slightly better.

Reply:
Here are some more options I've thought up:
- `allow-deleting-schema-locations-with-files`  doesn't make it clear that it's a fallback, but does make it clearer which files the toggle protects.
- `delete-schema-locations-when-checks-fail` makes it a bit clearer when the fallback occurs, but isn't super clear about what the "checks" in question are.
- `delete-schema-locations-with-files-when-checks-fail` is long, but clearer about what files it protects and that it's a fallback.

Reply:
It is not great but none of those are. Let's keep it.